qtpy/QtGui.py

```diff
@@ -8,14 +8,27 @@
 
 """Provides QtGui classes and functions."""
 
-from . import PYQT6, PYQT5, PYSIDE2, PYSIDE6
+from . import PYQT6, PYQT5, PYSIDE2, PYSIDE6, QtModuleNotInstalledError
+
+
+MISSING_OPENGL_NAMES = {}
+
 
 if PYQT5:
     from PyQt5.QtGui import *
 elif PYQT6:
     from PyQt6 import QtGui
     from PyQt6.QtGui import *
-    from PyQt6.QtOpenGL import *
```

Comment:
Sloppy copy and paste error, thanks for catching that!

Reply:
> you'll need to list the attributes present in PyQt/PySide's QtOpenGL 5.15 explicitly here.

Ah crap, that's much less fun... `qtpy/QtOpenGL.py` has a list of attributes it's importing from `QtGui`, so is it safe to assume this is the list of names we need to support?

This is kinda rough:

```py
try:
    from PySide6.QtOpenGL import *
except ImportError as error:
    names = [
        'QOpenGLBuffer',
        'QOpenGLFramebufferObject',
        'QOpenGLFramebufferObjectFormat',
        'QOpenGLShader',
        'QOpenGLShaderProgram',
        'QOpenGLContext',
        'QOpenGLContextGroup',
        'QOpenGLDebugLogger',
        'QOpenGLDebugMessage',
        'QOpenGLPixelTransferOptions',
        'QOpenGLTexture',
        'QOpenGLTextureBlitter',
        'QOpenGLVersionProfile',
        'QOpenGLVertexArrayObject',
        'QOpenGLWindow',
    ]
    for name in names:
        _missing_optional_names[name] = {
            'name': 'PySide6.QtOpenGL',
            'missing_package': 'pyopengl',
            'import_error': error,
        }
```

Reply:
Seems reasonable to me, though not the expert. If those lists are the same, instead of duplicating them maybe just stick them in a private module-level constant named, e.g. `_QTOPENGL_NAMES`, etc. and reference them in both?

Reply:
`_QTOPENGL_NAMES` is a good idea, consider it done.

I suppose I should audit `PyQt6.QtOpenGL` and `PySide6.QtOpenGL` and compare the available classes to this list.

Reply:
So I scraped `PyQt6/QtOpenGL.pyi` and `PySide6/QtOpenGL.pyi` and collected all the classes that start with `QOpenGL`. They both had some number of `QOpenGLFunctions_xxxx` classes, whichthat I discarded. 

## Comparison of the QtOpenGL classes to the list in `qtpy/QtOpenGL.py`:

<table>
<tr>
	<th>QtOpenGL.pyi
	<th>qtpy/QtOpenGL.py
<tr>
    <td>QOpenGLBuffer
    <td>QOpenGLBuffer
<tr>
    <td>
    <td>QOpenGLContext
<tr>
    <td>
    <td>QOpenGLContextGroup
<tr>
    <td>QOpenGLDebugLogger
    <td>QOpenGLDebugLogger
<tr>
    <td>QOpenGLDebugMessage
    <td>QOpenGLDebugMessage
<tr>
    <td>QOpenGLFramebufferObject
    <td>QOpenGLFramebufferObject
<tr>
    <td>QOpenGLFramebufferObjectFormat
    <td>QOpenGLFramebufferObjectFormat
<tr>
    <td>QOpenGLPaintDevice
    <td>
<tr>
    <td>QOpenGLPixelTransferOptions
    <td>QOpenGLPixelTransferOptions
<tr>
    <td>QOpenGLShader
    <td>QOpenGLShader
<tr>
    <td>QOpenGLShaderProgram
    <td>QOpenGLShaderProgram
<tr>
    <td>QOpenGLTexture
    <td>QOpenGLTexture
<tr>
    <td>QOpenGLTextureBlitter
    <td>QOpenGLTextureBlitter
<tr>
    <td>QOpenGLTimeMonitor
    <td>
<tr>
    <td>QOpenGLTimerQuery
    <td>
<tr>
    <td>QOpenGLVersionFunctionsFactory
    <td>
<tr>
    <td>QOpenGLVersionProfile
    <td>QOpenGLVersionProfile
<tr>
    <td>QOpenGLVertexArrayObject
    <td>QOpenGLVertexArrayObject
<tr>
    <td>QOpenGLWindow
    <td>QOpenGLWindow
</table>